benches/json.rs

```diff
@@ -35,6 +35,35 @@ fn boolean(input: &str) -> IResult<&str, bool> {
   alt((value(false, tag("false")), value(true, tag("true"))))(input)
 }
 
+fn u16_hex(input: &str) -> IResult<&str, u16> {
+  map_res(take(4usize), |s| u16::from_str_radix(s, 16))(input)
+}
+
+fn unicode_escape(input: &str) -> IResult<&str, char> {
```

Comment:
The standard library already implements decoding of UTF-16 sequences to characters. Is it possible to use [`char::decode_utf16()`](https://doc.rust-lang.org/std/char/fn.decode_utf16.html) instead of hand-rolling this code? I guess it would be a little awkward to construct an iterator over `u16`s decoded from the JSON. A simpler method might be something like:
```rust
// Assume unicode_escape() parses e.g. "\u12AB"

fn unicode_char(input: &str) -> IResult<&str, char> {
  let (input, cp) = unicode_escape(input)?;
  if let Some(c) = char::from_u32(cp) {
    return Ok((input, c));
  }

  map_res(unicode_escape, |cp2| {
    char::decode_utf16([cp, cp2].iter().copied()).next().unwrap()
  })(input)
}
```